src/cirrus/lib2/utils.py

```diff
@@ -168,9 +168,10 @@ def extract_record(record):
     if "Message" in record:
         record = json.loads(record["Message"])
 
-    if "url" not in record and "Parameters" in record and "url" in record["Parameters"]:
+    if "url" not in record and "Parameters" in record:
         # this is Batch, get the output payload
-        record = {"url": record["Parameters"]["url"].replace(".json", "_out.json")}
+        if "url_out" in record["Parameters"]:
+            record = {"url": record["Parameters"]["url_out"]}
```

Comment:
So, I could be wrong here, but for the avoidance of a breaking change, I'd recommend issuing a deprecation warning, and supporting the legacy behavior
```suggestion
            record = {"url": record["Parameters"]["url_out"]}
        else:
            record = {"url": record["Parameters"]["url"].replace(".json", "_out.json")}
            import warnings
            warnings.warn("Using the deprecated implicit '_out.json' url-hack",
                          category=DeprecationWarning, stacklevel=2)
```

Reply:
I was wondering if I should have this be backwards compatible, but then thought that if you upgrade to this version you are updating both the pre and post batch lambdas...so is there a case where `url_out` would not be present?

Reply:
Oh, I see. pre-batch uses the same WKloc, so it does work w/o changing the batch task, but the task should still be changed.